lib/Caxy/HtmlDiff/ListDiff.php

```diff
@@ -0,0 +1,366 @@
+     */
+    protected function processPlaceholders($text, array $matches)
+    {
+        // Prepare return
+        $returnText = array();
+        // Save the contents of all list nodes, new and old.
+        $contentVault = array(
+            'old' => $this->getListContent('old', $matches),
+            'new' => $this->getListContent('new', $matches)
+        );
+        
+        $count = 0;
+        // Loop through the text checking for placeholders. If a nested list is found, create a new ListDiff object for it.
+        foreach (explode(' ', $text) as $word) {
+            $preContent = $this->checkWordForDiffTag($this->stripNewLine($word));
+            
+            if (in_array(
+                    is_array($preContent) ? $preContent[1] : $preContent, 
+                    $this->isolatedDiffTags
+                )
+            ) {
+                $oldText = implode('', $contentVault['old'][$count]);
+                $newText = implode('', $contentVault['new'][$count]);
+                $content = $this->diffList($oldText, $newText);
+                $count++;
+            } else {
+                $content = $preContent;
+            }
+            
+            $returnText[] = is_array($preContent) ? $preContent[0] . $content . $preContent[2] : $content;
+        }
+        // Return the result.
+        return implode(' ', $returnText);
+    }
+    
+    protected function checkWordForDiffTag($word)
+    {
+        foreach ($this->isolatedDiffTags as $diffTag) {
+            if (strpos($word, $diffTag) > -1) {
+                $position = strpos($word, $diffTag);
+                $length = strlen($diffTag);
+                $result = array(
+                    substr($word, 0, $position),
+                    $diffTag,
+                    substr($word, ($position + $length))
+                );
+                
+                return $result;
+            }
+        }
+        
+        return $word;
+    }
+    
+    /**
+     * Used to remove new lines.
+     */
+    protected function stripNewLine($text)
+    {
+        return trim(preg_replace('/\s\s+/', ' ', $text));
+    }
+    
+    /**
+     * Grab the list content using the listsIndex array.
+     */
+    protected function getListContent($indexKey = 'new', array $matches)
+    {
+        $bucket = array();
+        $start = $this->listsIndex[$indexKey][$matches[$indexKey]];
+        $stop = $this->listsIndex[$indexKey][$matches[$indexKey] + 1];
+        for ($x = $start; $x < $stop; $x++) {
+            if (in_array($this->list[$indexKey][$x], $this->isolatedDiffTags)) {
+                $bucket[] = $this->listIsolatedDiffTags[$indexKey][$x]; 
+            }
+        }
+        return $bucket;
+    }
+    
+    /**
+     * indexLists
+     * 
+     * Index the list, starting positions, so that we can refer back to it later.
+     * This is used to see where one list node starts and another ends.
+     */
+    protected function indexLists()
+    {
```

Comment:
This function would have a similar issue to what was mentioned in the [comment below](https://github.com/caxy/php-htmldiff/pull/14/files#r40692297) -- if there is a nested list item that is not nested in an ol, dl, or ul.